### PR TITLE
fix(core): Use owners file to export wf owners

### PR DESCRIPTION
**Pull request by flipswitchingmonkey**

### CODE REVIEW

**Review by krynble — CHANGES_REQUESTED** [review]

File: packages/cli/src/environments/sourceControl/sourceControl.service.ee.ts [hunk 1]
```diff
@@ -326,6 +330,13 @@ export class SourceControlService {
 					id = workflow.id;
 					name = workflow.name;
 				}
+				const existingWorkflow = await Db.collections.Workflow.find({
```
[the remark this answers is not in the document]
Comment (flipswitchingmonkey) [reply packages/cli/src/environments/sourceControl/sourceControl.service.ee.ts:333]: good point

File: packages/cli/src/environments/sourceControl/sourceControlImport.service.ee.ts [hunk 2]
```diff
@@ -273,14 +274,44 @@ export class SourceControlImportService {
...
+					ownerRecords[workflowId] =
+						existingUsers.find((e) => e.email === ownerEmails[workflowId])?.id ?? userId;
+				});
+			}
+		}
+
+		console.log(ownerRecords);
+
+		let importWorkflowsResult = new Array<{ id: string; name: string } | undefined>();
 		await Db.transaction(async (transactionManager) => {
```
[the remark this answers is not in the document]
Comment (flipswitchingmonkey) [reply packages/cli/src/environments/sourceControl/sourceControlImport.service.ee.ts]: removed transactions

File: packages/cli/src/environments/sourceControl/sourceControlExport.service.ee.ts [hunk 3]
```diff
@@ -186,6 +190,11 @@ export class SourceControlExportService {
 			const removedFiles = await this.rmDeletedWorkflowsFromExportFolder(sharedWorkflows);
 			// write the workflows to the export folder as json files
 			await this.writeExportableWorkflowsToExportFolder(sharedWorkflows);
+			// write list of owners to file
+			const ownersFileName = this.getOwnersPath();
+			const owners: Record<string, string> = {};
+			sharedWorkflows.forEach((e) => (owners[e.workflowId] = e.user.email));
```
Comment (krynble) [inline packages/cli/src/environments/sourceControl/sourceControlExport.service.ee.ts:196]: One problem I see here is that we're always sending the full file, regardless of the workflows actually exported.

One problem I had is that I have over 100 workflows, but I exported a single one. My owners file contains all the 100 workflow owner names, even though my repo only contains a single workflow file.

I'm thinking that maybe we should read the file contents and append the rows that were added WDYT?

**Review by krynble — CHANGES_REQUESTED** [review]
Other than the comment I left, it all looks fine.

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.236.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.236.0)

**Pull request merged**